### PR TITLE
Better WebMC

### PR DESCRIPTION
**Pull request by KaffinPX**
Adds basic daycycle
Prevents dropping when a menu is opened
Better ESC handling like Minecraft (now its normal webmc :()

### CODE REVIEW

**Comment by KaffinPX** [comment]
Note: daycycle isnt tested

**Review by michaljaz — COMMENTED** [review]
it is not working :(

**Comment by KaffinPX** [comment]
Oh lol wait a minute i forget pulling my latest changes

File: src/scripts/EventHandler.js [hunk 1]
```diff
@@ -61,10 +61,16 @@ class EventHandler {
...
-        this.game.socket.emit('drop')
+        if (this.gameState !== 'chat' && this.gameState !== 'menu') {
+          this.game.socket.emit('drop')
+        }
       }
-      if (z.code === 'Escape' && this.gameState === 'inventory') {
-        this.setState('menu')
+      if (z.code === 'Escape') {
+        if (this.gameState === 'inventory' || this.gameState === 'chat' || this.gameState === 'menu') {
+          this.setState('game')
```
Comment (misioxd) [inline src/scripts/EventHandler.js:70]: Escape can't redirect to game because it exits pointerlock

Comment (michaljaz) [inline src/scripts/EventHandler.js:70]: Escape code cannot redirect to game because it exits pointerlock

Comment (KaffinPX) [reply src/scripts/EventHandler.js:70]: Oh wait a minute

Comment (KaffinPX) [reply src/scripts/EventHandler.js:70]: Reverting to original esc lol

**Comment by michaljaz** [comment]
day cycle is bad implemented
it doesn't look good

**Comment by KaffinPX** [comment]
bruh rly nevermind

**Pull request closed without merging**